### PR TITLE
Add debug option.

### PR DESCRIPTION
**Pull request by 130s**
Undefined property error is hard to debug without knowing which properties are read / are not read. So I'm adding debug feature.

```
rosrun xacro xacro foo.xacro --debug

2016-07-20 17:57:34,024 - DEBUG - Input xacro file full path=/home/roslinguini/src/sirobot1_gazebo/model/foo.xacro
2016-07-20 17:57:34,119 - DEBUG - name=pi, val=3.1415926535897931
2016-07-20 17:57:34,119 - DEBUG - name=small_mass, val=0.001
2016-07-20 17:57:34,120 - DEBUG - name=pi, val=3.1415926535897931
2016-07-20 17:57:34,120 - DEBUG - name=pi, val=3.1415926535897931
2016-07-20 17:57:34,120 - DEBUG - name=width_shelf, val=0.005
2016-07-20 17:57:34,120 - DEBUG - name=radius, val=0.002
2016-07-20 17:57:34,120 - DEBUG - name=pi, val=3.1415926535897931
2016-07-20 17:57:34,120 - DEBUG - name=pi, val=3.1415926535897931
2016-07-20 17:57:34,121 - DEBUG - name=wheel_mass, val=0.5
2016-07-20 17:57:34,121 - DEBUG - name=sphere_mass, val=0.3
```

### CODE REVIEW

**Comment by rhaschke** [comment]
You should consider to switch to the new xacro version of >= Jade, which is also backported to Indigo and available via new option --inorder. This new version has extensive debugging features with options `-v` ... `-vvv`. IMHO the Indigo branch is deprecated.

**Comment by 130s** [comment]
tldr; Closing since a workaround is provided that should work great (only offline for Indigo >= though).

> You should consider to switch to the new xacro version of >= Jade, which is also backported to Indigo and available via new option --inorder. 

I guess you're suggesting to check offline the xacro file using xacro from >= Jade distro. I see the point after looking at the wiki http://wiki.ros.org/xacro#Processing_Order

`inorder` isn't available during runtime on Indigo => distro. I guess this feature will never be available there?

> IMHO the Indigo branch is deprecated.

I don't know if that is acceptable as an end-user's perspective, since ROS Indigo itself is supported until 2019...

**Comment by rhaschke** [comment]
> > IMHO the Indigo branch is deprecated.
> 
> I don't know if that is acceptable as a end-user since ROS Indigo itself is supported until 2019...

Of course ROS Indigo will be further supported and bug fixes will be performed.
However, you are requesting a new feature for Indigo ;-)

**Comment by 130s** [comment]
> > > ```
> > >    IMHO the Indigo branch is deprecated.
> > > ```
> > 
> > ```
> > I don't know if that is acceptable as a end-user since ROS Indigo itself is supported until 2019...
> > ```
> 
> Of course ROS Indigo will be further supported and bug fixes will be performed.
> However, you are requesting a new feature for Indigo ;-)

I believe that it is NOT commonly accepted idea that new feature won't be added to older distribution (e.g. Indigo).
As long as a distro is supported and new feature development is happening on its repository, I think it's natural for many users to expect that they'll keep getting those features.

I'm not forcing xacro maintainers to backport new features into Indigo, but I just wanted to point that out.
